engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobEventProcessors.java

```diff
@@ -12,16 +12,19 @@
 import io.camunda.zeebe.engine.processing.common.EventHandle;
 import io.camunda.zeebe.engine.processing.streamprocessor.TypedRecordProcessors;
 import io.camunda.zeebe.engine.processing.streamprocessor.writers.Writers;
+import io.camunda.zeebe.engine.state.immutable.ScheduledTaskState;
 import io.camunda.zeebe.engine.state.mutable.MutableProcessingState;
 import io.camunda.zeebe.protocol.record.ValueType;
 import io.camunda.zeebe.protocol.record.intent.JobBatchIntent;
 import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import java.util.function.Supplier;
 
 public final class JobEventProcessors {
 
   public static void addJobProcessors(
       final TypedRecordProcessors typedRecordProcessors,
       final MutableProcessingState processingState,
+      final Supplier<ScheduledTaskState> scheduledTaskStateSupplier,
```

Comment:
🔧 As this supplier is taking the role of a factory, let's call it so (in other places, it's called `scheduledTaskStateFactory`)